### PR TITLE
Add pull request (PR) template

### PR DESCRIPTION
**Pull request by josh-wong**
## Description

This PR updates the PR template in this repository. The template is for helping to provide context for people who are involved with PRs and for future reference.

## Related issues and/or PRs

N/A

## Changes made

Created the pull request template in the **.github** folder.

## How these changes were tested

Checked that this PR template looks as expected in GitHub.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+> Provide any additional information or notes that may be relevant to the reviewers or stakeholders.
+
+## Release notes
+
+> Provide release note text for this PR based on the guidelines at [Guidelines for Writing Release Notes](https://developers.scalar-labs.com/docs/style-guide/release-notes/).
+> 
+> For example:
+> 
+> Added validation for the Consensus Commit mutation operation.
```
Comment (kota2and3kan) [inline .github/pull_request_template.md:38]:
```suggestion
```

@ypeckstadt @thongdk8 
I think we don't have any release notes for the Helm Chart of ScalarFlow. Is my understanding correct?
If yes, I think we can remove this section at this time.
However, if you have any plans to create release notes in the future, I think it's OK to remain this section.

What do you think?

**Review by kota2and3kan — COMMENTED** [review]
Overall looks good to me, but I have one suggestion.

@ypeckstadt @thongdk8
This is a repository for Helm Chart of ScalarFlow.
Could you please review this PR?

File: .github/pull_request_template.md (hunk 1 above)
Comment (thongdk8) [reply .github/pull_request_template.md:34]: Yes, currently we dont have but we will add it anyways, so it is fine to keep the above section I think.

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

File: .github/pull_request_template.md (hunk 1 above)
Comment (ypeckstadt) [reply .github/pull_request_template.md:34]: Yeah, I think it is better to keep it as in the end ScalarFlow helm charts will be public. So there will be release notes.

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**